### ISSUE TITLE
@requireAll does not work with @type

### ISSUE THREAD

**Issue by ioggstream** [digitalbazaar/pyld#165]
## I expect

```
frame = {
"@type": {
  @requireAll: true,
  @value: [ skos:Concept, dct:Location ]
}
}
```

## Instead

E           Cause: ('Invalid JSON-LD syntax; "@type" value must be a string, an array of strings, or an empty object.',)

## Trace

```
E           pyld.jsonld.JsonLdError: ('Could not expand frame before framing.',)
E           Type: jsonld.FrameError
E           Cause: ('Invalid JSON-LD syntax; "@type" value must be a string, an array of strings, or an empty object.',)
E           Type: jsonld.SyntaxError
E           Code: invalid type value
E           Details: {'value': {'@value': ['http://www.w3.org/2004/02/skos/core#Concept', 'http://publications.europa.eu/ontology/authority/Country', 'http://purl.org/dc/terms/Location']}}  File "~/.local/lib/python3.9/site-packages/pyld/jsonld.py", line 1043, in frame
E               expanded_frame = self.expand(frame, opts)
E             File "~/.local/lib/python3.9/site-packages/pyld/jsonld.py", line 870, in expand
E               expanded = self._expand(active_ctx, None, document, options,
E             File "~/.local/lib/python3.9/site-packages/pyld/jsonld.py", line 2336, in _expand
E               self._expand_object(
E             File "~/.local/lib/python3.9/site-packages/pyld/jsonld.py", line 2542, in _expand_object
E               _validate_type_value(value, options.get('isFrame'))
E             File ~/.local/lib/python3.9/site-packages/pyld/jsonld.py", line 6361, in _validate_type_value
```

**Comment by mielvds** [comment]
@ioggstream do you have any proper input and frame data to test this with? The given frame is not valid JSON. Python or JSON-LD. 

The `@requireAll` flag only works for properties afaik.

**Comment by ioggstream** [comment]
Hi @mielvds , I just updated the issue. Currently, jsonld `@type` matches "any" `@type`. I'd like a way to match "all" the `@type`s.

I don't know if there's an interoperable way to do this though.

cc: @pchampin

**Comment by pchampin** [comment]
`@requireAll` is not what you are looking for; it is not about matching all values for a given property, but matching all [properties for a given frame](https://www.w3.org/TR/json-ld11-framing/#dfn-require-all-flag). That's why you get an error: `@requireAll` can not be set inside a property value, only in the frame object itself.

For each property, you are right, a node object matches if *any* value in the frame matches one value of the object. AFAIK, there is no way to turn that into an *all*.

To summarize: the title of this issue is misled. `@requireAll` works with `@type` just as well as any other property. The error you get is not caused by `@type` (replace `@type` with `another:property` in your frame, you still get the same error). It is caused by you misplacing the `@requireAll` keyword in the frame.

This is *not* a bug in PyLD.

**Comment by mielvds** [comment]
Thanks for the insights @pchampin! 🙏 Closing this, assuming that the question has been answered.

**Comment by ioggstream** [comment]
Thanks @pchampin!

Should I file a feature request somewhere?

**Comment by pchampin** [comment]
> Thanks [@pchampin](https://github.com/pchampin)!
> 
> Should I file a feature request somewhere?

yes: https://github.com/w3c/json-ld-framing/issues/
